addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -32,11 +25,15 @@ def _compute_timesheet_generate(self):
         for leave_type in self:
             leave_type.timesheet_generate = leave_type.timesheet_task_id and leave_type.timesheet_project_id
 
+    @api.depends('company_id')
+    def _compute_timesheet_project_id(self):
+        for leave in self:
+            leave.timesheet_project_id = leave.company_id.internal_project_id or False
+
     @api.depends('timesheet_project_id')
     def _compute_timesheet_task_id(self):
         for leave_type in self:
-            company = leave_type.company_id if leave_type.company_id else self.env.company
-            default_task_id = company.leave_timesheet_task_id
+            default_task_id = leave_type.company_id.leave_timesheet_task_id
```

Comment:
```suggestion
            default_task_id = leave_type.company_id and leave_type.company_id.leave_timesheet_task_id
```
Same as above

Reply:
Hello @MissingNoShiny , Thank you for looking into it :slightly_smiling_face: 

Although leave.company_id is none, leave.company_id.internal_project_id will also become none and be set as none. so, when the leave type lacks a company, project_id will get a value of none.

I think we don't need False or and condition on it.